### PR TITLE
Interpret two braces in the interpolated string

### PR DESCRIPTION
**Pull request by ureishi**
Interpret two braces ("{{" or "}}") as one brace ("{" or "}") in the interpolated string.

cf. https://docs.microsoft.com/en-us/dotnet/standard/base-types/composite-formatting#escaping-braces

### CODE REVIEW

**Comment by ureishi** [comment]
I have read the CLA Document and I hereby sign the CLA

File: Assets/UdonSharp/Tests~/TestScripts/BugTests/StringInterpolation/StringInterpolation.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+            tester.TestAssertion("String interpolation alignment clause", $"{12345,6}" == " 12345");
+
+            tester.TestAssertion("String interpolation format clause", $"{12345:D06}" == "012345");
+
+            tester.TestAssertion("String interpolation alignment & format clause", $"{12345,7:D06}" == " 012345");
+
+            tester.TestAssertion("String interpolation empty", $"" == string.Empty);
+
+            tester.TestAssertion("String interpolation null", $"{null}" == string.Empty);
```
Comment (bdunderscore) [inline Assets/UdonSharp/Tests~/TestScripts/BugTests/StringInterpolation/StringInterpolation.cs:42]: It's probably a good idea to have a separate test which uses a constant null variable as well, to ensure that we test the case where the null isn't eliminated by the string formatting expression logic before execution.

**Comment by MerlinVR** [comment]
Okay I was confused for a moment here. The relevant tests for this change are 
```cs
tester.TestAssertion("String interpolation 1 left brace", $"One left brace \"{{\"" == "One left brace \"{\"");
tester.TestAssertion("String interpolation 1 right brace", $"One right brace \"}}\"" == "One right brace \"}\"");
tester.TestAssertion("String interpolation multi braces", $"{{1}}, {{{{2}}}}, {{{{{{3}}}}}}" == "{1}, {{2}}, {{{3}}}");
```

where no interpolation takes place and the compiler optimizes them to const strings but did not unescape the string. The rest just provide better coverage for string interpolation testing.

**Comment by ureishi** [comment]
I will limit it to test cases where parentheses are escaped in the string interpolation, to make the purpose of the test clearer. The two candidates are as follows:
```cs
tester.TestAssertion("String interpolation braces escape 1", $"}}, {{}}, {{" == "}, {}, {");

tester.TestAssertion("String interpolation braces escape 2", $"{{{1}}}, {{{{{2}}}}}, {{{{{{{3}}}}}}}" == "{1}, {{2}}, {{{3}}}");
```
I thought it appropriate to prepare a detailed test on string interpolation as a separate test, if necessary.

**Comment by MerlinVR** [comment]
It's okay to have the extra tests, I was just clarifying for future reference.

**Comment by ureishi** [comment]
> It's okay to have the extra tests, I was just clarifying for future reference.

Okay, but I pushed before reading the message.... What should I do, leave it as it is, push the previous test case again, or create a new file as a detailed test case about string interpolation?

**Comment by MerlinVR** [comment]
It'd be good to have the extra tests just in case if you want to re-add them since you already have them written.

**Pull request merged**